src/commands/deploy/deploy.ts

```diff
@@ -88,6 +95,26 @@ async function deploy(actionContext: IActionContext, arg1: vscode.Uri | string |
     await notifyDeployComplete(context, node, context.workspaceFolder.uri.fsPath);
 }
 
+async function updateWorkerProcessTo64BitIfRequired(context: IDeployContext, siteConfig: WebSiteManagementModels.SiteConfigResource, node: SlotTreeItemBase, language: ProjectLanguage): Promise<void> {
+    const functionProject: string | undefined = await tryGetFunctionProjectRoot(context.workspaceFolder.uri.fsPath);
+    if (functionProject === undefined) {
+        throw new Error(localize('failedToFindFuncHost', 'Unable not locate Azure Functions project.'));
+    }
+    const projectFiles: string[] = await dotnetUtils.getProjFiles(language, functionProject);
+    if (projectFiles.length === 0) {
+        throw new Error(localize('unableToFindProj', 'Unable to detect project file.'));
+    }
+    const mainProject: string = path.join(functionProject, projectFiles[0]);
+    const platformTarget: string | undefined = await dotnetUtils.getPlatformTarget(mainProject);
+    if (platformTarget === 'x64' && siteConfig.use32BitWorkerProcess === true) {
+        const config: WebSiteManagementModels.SiteConfigResource = {
+            use32BitWorkerProcess: false
+        };
+        context.stopAppBeforeDeploy = true;
+        await node.root.client.updateConfiguration(config);
```

Comment:
We try to avoid silent changes to a user's function app. Please add a warning similar to this: https://github.com/microsoft/vscode-azurefunctions/blob/d1318cad6809dbe65029dfe2012c76a1b17d0df5/src/commands/deploy/verifyAppSettings.ts#L86

For example, maybe they didn't actually want 64-bit and they will cancel this deploy and change their local project instead of the app in Azure.

Reply:
Acknowledged